scipy/stats/tests/test_stats.py

```diff
@@ -6052,16 +6071,19 @@ def test_normalitytests():
     # Test axis=None (equal to axis=0 for 1-D input)
     assert_array_almost_equal(stats.normaltest(x, axis=None),
                               (st_normal, pv_normal))
-    assert_array_almost_equal(stats.skewtest(x, axis=None),
-                              (st_skew, pv_skew))
+    res = stats.skewtest(x_xp, axis=None)
+    xp_assert_close(res.statistic, st_skew)
+    xp_assert_close(res.pvalue, pv_skew)
     assert_array_almost_equal(stats.kurtosistest(x, axis=None),
                               (st_kurt, pv_kurt))
 
-    x = np.arange(10.)
-    x[9] = np.nan
+    x = xp.arange(10.)
+    x[9] = xp.nan
     with np.errstate(invalid="ignore"):
-        assert_array_equal(stats.skewtest(x), (np.nan, np.nan))
+        assert_array_equal(stats.skewtest(x), (xp.nan, xp.nan))
 
+    # nan_policy only compatible with NumPy arrays
+    x = np.asarray(x)
```

Comment:
As noted earlier this is slightly wasteful but planning to look at converting the other normal tests too, after which I would like to look at breaking this test up into slightly more manageable chunks